domains/certificates/Query.API/API/DataSyncSyncer/DataSyncSyncerWorker.cs

```diff
@@ -13,41 +18,58 @@ namespace API.DataSyncSyncer;
 internal class DataSyncSyncerWorker : BackgroundService
 {
     private readonly ILogger<DataSyncSyncerWorker> logger;
-    private readonly IEventStore eventStore;
-    private readonly string? gsrn;
+    private readonly IIntegrationEventBus integrationEventBus;
+    private readonly IDataSync dataSync;
+    private readonly List<MasterData> masterData;
 
-    public DataSyncSyncerWorker(ILogger<DataSyncSyncerWorker> logger, IEventStore eventStore,
-        MockMasterDataCollection collection)
+    public DataSyncSyncerWorker(
+        ILogger<DataSyncSyncerWorker> logger,
+        MockMasterDataCollection collection,
+        IIntegrationEventBus queue,
+        IDataSync dataSync
+    )
     {
         this.logger = logger;
-        this.eventStore = eventStore;
-        var masterData = collection.Data.FirstOrDefault();
-        gsrn = masterData?.GSRN ?? null;
+        masterData = collection.Data.ToList();
+        integrationEventBus = queue;
+        this.dataSync = dataSync;
     }
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        if (string.IsNullOrWhiteSpace(gsrn))
-        {
-            logger.LogWarning("No master data loaded. Will not produce any events");
-        }
-
-        await Task.Delay(TimeSpan.FromMilliseconds(500),
-            stoppingToken); //allow application to start before producing events
-        var random = new Random();
 
         while (!stoppingToken.IsCancellationRequested)
         {
-            if (!string.IsNullOrWhiteSpace(gsrn))
+            var period = GetPeriod();
+
+            foreach (var data in masterData.Where(it => !string.IsNullOrWhiteSpace(it.GSRN)))
             {
+                var measurement = await dataSync.GetMeasurement(data.GSRN, period);
+                await integrationEventBus.Produce(stoppingToken, measurement);
```

Comment:
I think this principle will not work in production. On each "timer tick" we are asking for measurements from the last hour. But I do not think DataHub will have data ready that fast. 
So the principle should more be that it fetches data from the last time that metering point received measurements on each "timer tick".